scipy/signal/_signaltools.py

```diff
@@ -93,6 +93,18 @@ def _inputs_swap_needed(mode, shape1, shape2, axes=None):
     return not ok1
 
 
+def _reject_objects(arr, name):
+    """Warn if arr.dtype is object or longdouble or float16.
+    """
+    dt = np.asarray(arr).dtype
+    if not (np.issubdtype(dt, np.integer)
+            or dt in [np.bool_, np.float32, np.float64, np.complex64, np.complex128]
+    ):
+        warnings.warn(f"dtype={dt} is not supported by {name}",
```

Comment:
Do we want to commit to when this will raise an error?
```suggestion
        warnings.warn(f"dtype={dt} is not supported by {name} and will raise an error in SciPy 1.17.0",
```

Reply:
Can we also mention the supported dtypes in the warning message.

Reply:
> Can we also mention the supported dtypes in the warning message.

LGTM once this is addressed! (generally good to point to a workaround / correct use in error messages when possible)